src/main/resources/lib/guillotine/subscription/event.js

```diff
@@ -27,17 +28,52 @@ function createEventObjectType(context) {
 }
 
 function createNewEventPublisher(context) {
+    const siteRepo = ctxLib.get().repository;
+
     const processor = graphQlRxLib.createPublishProcessor();
+
     eventLib.listener({
         type: '*',
         callback: function (event) {
-            if (!(event.type == 'application' && event.data.applicationKey == app.name)) { //TODO Temporary Workaround
-                processor.onNext(event);
-            }
+            context.options.subscriptionEventTypes.some(subscriptionEventType => {
+                if (subscriptionEventType === '*') {
+                    processor.onNext(event);
+                    return true;
+                } else {
+                    let eventType = subscriptionEventType.endsWith('*')
+                                    ? subscriptionEventType.substring(0, subscriptionEventType.length - 1)
+                                    : subscriptionEventType;
+
+                    let anyMatched = false;
+
+                    if (event.type.startsWith(eventType)) {
+                        anyMatched = anyFilterMatch(event, siteRepo, event.type);
+                    } else if (subscriptionEventType === event.type) {
+                        anyMatched = anyFilterMatch(event, siteRepo, subscriptionEventType);
+                    }
+
+                    if (anyMatched) {
+                        processor.onNext(event);
+                        return true;
+                    }
+                }
+            });
         }
     });
     return processor;
 }
 
+function anyFilterMatch(event, siteRepo, type) {
```

Comment:
This is a confusing name. Which of  _any_ filters this function matches?

Reply:
done